lib/stages/RfiFrameDrop.hpp

```diff
@@ -86,16 +87,16 @@ class RfiFrameDrop : public kotekan::Stage {
     Buffer* _buf_out;
 
     /// Thresholds
-    std::vector<std::tuple<float, size_t, float>> _thresholds;
+    std::shared_ptr<std::vector<std::tuple<float, size_t, float>>> _thresholds;
```

Comment:
You might consider binding both `_sk_exceeds` and `thresholds` into a single new struct, then you only need a single shared_ptr to point to all the threshold related state. That would eliminate any theoretical potential of inconsistency.